### DISCUSSION TITLE
I would like to know how to use Vercel's Serverless Function instead of Cors Proxy

### DISCUSSION THREAD

**Discussion by BJS5705** [gamer2810/steam-miniprofile#20] [Q&A] [answered]
As you know, The Online/Offline status update is little bit slow. Sometimes it takes over 10 minutes. So to solve this CORS Proxy's delay problem, I had to fork your project and deploy to Vercel. Because Vercel supports Serverless Function so the mini profile request could be faster and more accurate. But for some reason it doesn't work.

### (api/steamProfile.js)

```javascript
export default async function handler(req, res) {
  const { steamId, language } = req.query;
  const url = `https://steamcommunity.com/miniprofile/${steamId}?l=${language || 'koreana'}`;

  if (req.method === 'OPTIONS') {
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
    return res.status(204).end();
  }

  try {
    const response = await fetch(url);
    const data = await response.text();

    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    res.status(200).send(data);
  } catch (error) {
    console.error('Error fetching data:', error);
    res.status(500).json({ error: 'Error fetching data' });
  }
}
```

### (index.html)
```<html lang="en">
<head>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="robots" content="noindex, nofollow">
  <meta name="referrer" content="no-referrer">
  <link async rel="preload" as="style"
        href="https://community.cloudflare.steamstatic.com/public/shared/css/shared_global.css"
        onload="this.onload=null;this.rel='stylesheet'">
  <noscript>
    <link rel="stylesheet" href="https://community.cloudflare.steamstatic.com/public/shared/css/shared_global.css">
  </noscript>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"
          integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg=="
          crossorigin="anonymous" referrerpolicy="no-referrer"></script>
  <script defer async>
    $(function () {
      let searchParams = new URLSearchParams(window.location.search);
      let steamId = searchParams.get('accountId') ? searchParams.get('accountId') : '76561198055079301';
      let interactive = searchParams.get('interactive') ? searchParams.get('interactive') : false;
      let vanityId = searchParams.get('vanityId') ? searchParams.get('vanityId') : 'intheheaven';
      let language = searchParams.get('lang') ? searchParams.get('lang') : 'koreana';
      let corsURL = `https://myvercelurlexample/api/steamProfile?steamId=${steamId}&language=${language}`;

      ///rest of codes are same
```
In Vercel's deployment logs, it shows the result that steamProxy.js got respond "200" perfectly, but I couldn't get any Steam profile information. Just only mini-profile's default frame. No nickname, no badges, no level, no on/offline status.

I don't know how your corsUrl works exactly. I think serverless function is not the problem. Because some how, steamProxy.js getting respond 200 anyway. But index.html seems cannot bring mini-profile html information for some reason. Or just there's another problem.
My English is really bad so it might be sound like rude. Sorry.

**Comment by gamer2810** [comment]
Can you check that your `data` variable in `steamProxy.js` correctly contains the whole HTML content of the steam miniprofile? 

Also I do not think that the delay in Online/Offline status update is due to the proxy. 
The actual Steam's `website` (i.e. `https://steamcommunity.com/id/yourSteamId`) does not update your presence that quick. 
This is not the case with the Steam application, it updates the presence instantly.
My guess is the application is using some sort of push notifications to override the presence, in which case you need to find the script responsible for that mechanism and include it in your `steam-miniprofile` fork. 

I'm sorry for the late reply, I did not see the notification about the discussions 🙏 .

**Reply by BJS5705** [reply]
Sorry for keep bothering you.
Do ```steam-proxy.js``` must contains steam miniprofile's html content? Seems your project also doesn't contain html information. It just brings css data from code down below.
```
<noscript>
<link rel="stylesheet" href="https://community.cloudflare.steamstatic.com/public/shared/css/shared_global.css">
</noscript>
```
My ```steam-proxy.js``` contains code above too. So it should bring steam miniprofile data from there like your project does. But it doesn't. My coding skill is sucks, if you don't mine I want you to help me out.🙏
My github repository url is [here](https://github.com/BJS5705/Mini-Profile-Vercel-Proxy) and Verce's app url is https://mini-profile-vercel-proxy.vercel.app/
Can you check codes and tell me what parts are wrong? The reason that I use Vercel is because CORS proxy has its own delay or maximun limitations for any request compared with Steam API request. I created widget that checks my Steam account's presence through requesting Steam API. And It updates Online/Offline status right away every API request, in the other hand miniprofile doesn't.

I have no idea how to replace CORS Proxy to Steam API request only for On/Offline part.

If this problem get solved, I can probably make On/Offline section that works through Steam API only for that part.

**Reply by gamer2810** [reply, marked as answer]
I see the problem now.
Your code is working fine.

My proxy server has a mechanism to translate different types of ID into Steam's ID. 
Yours does not have that mechanism so it could not find your profile.
Just use your normal `AccountID` (which can be found here [https://steamdb.info/calculator/](https://steamdb.info/calculator/)) and you should be fine:

`https://mini-profile-vercel-proxy.vercel.app/api/steam-proxy?accountId=94813573&lang=koreana`

If you would like to see the ID translation code I can get that and explain how it works to you.

Have a nice day ☕ and please mark this as answer if it answered your question.
